src/ast/codegen_llvm.cpp

```diff
@@ -177,6 +177,38 @@ void CodegenLLVM::visit(Builtin &builtin)
     expr_ = b_.CreateLoad(dst);
     b_.CreateLifetimeEnd(dst);
   }
+  else if (!builtin.ident.compare(0, 4, "sarg") && builtin.ident.size() == 5 &&
+      builtin.ident.at(4) >= '0' && builtin.ident.at(4) <= '9')
+  {
+    int arg_num = atoi(builtin.ident.substr(4).c_str());
+    if (probetype(current_attach_point_->provider) == ProbeType::usdt) {
+      expr_ = b_.CreateUSDTReadArgument(ctx_, current_attach_point_,
+                                        arg_num, builtin, bpftrace_.pid_);
+      return;
+    }
+
+    int sp_offset = arch::offset("sp");
+    if (sp_offset == -1)
+    {
+      std::cerr << "negative offset for stack pointer" << std::endl;
+      abort();
+    }
+
+    AllocaInst *dst;
+    Value *src;
+
+    dst = b_.CreateAllocaBPF(SizedType(Type::integer, 8), "reg_sp");
```

Comment:
You should be able to do this without a proberead, see 7d0aa993433dd8b743898c8e1743d72b5af75684 which was merged last night.